spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteTablePathsAction.java

```diff
@@ -923,16 +1005,20 @@ protected void checkFileNum(
             .load(result.fileListLocation())
             .as(Encoders.STRING())
             .collectAsList();
-    assertThat(filesToMove.stream().filter(f -> f.endsWith(".metadata.json")).count())
-        .withFailMessage("Wrong rebuilt version file count")
+    Predicate<String> isManifest = f -> (f.endsWith("-m0.avro")) || (f.endsWith("-m1.avro"));
```

Comment:
I think these may be great changes, but are they necessary for this?  What do you think to make a separate pr for these changes and have some explanation there?  It would make this specific change easier to review.

Reply:
Sorry about the confusion, yeah I think this might be needed for testing out the change. I realize that the overwrite of unpartitioned table generate 2 manifests where `m0.avro` contains the deleted entry of old data file and `m1.avro` contains the appended entry for new data file. So previous we only consider file name ends up with `m0.avro` as manifest count might be a off. 

Also moved away from `withFailMessage()` to `as()` to provide a little bit more context incase of assertion failure.